sql/src/main/java/io/crate/executor/transport/AlterTableOperation.java

```diff
@@ -125,50 +126,50 @@ public void onFailure(Throwable e) {
 
             if (analysis.partitionName().isPresent()) {
                 String index = analysis.partitionName().get().asIndexName();
-                results.add(updateMapping(analysis.tableParameter().mappings(), index));
-                results.add(updateSettings(parameterWithFilteredSettings, index));
+                updateMapping(analysis.tableParameter().mappings(), result, operations, index);
+                updateSettings(parameterWithFilteredSettings, result, operations, index);
             } else {
-                // template gets all changes unfiltered
-                results.add(updateTemplate(analysis.tableParameter(), table.ident()));
-
                 if (!analysis.excludePartitions()) {
-                    // resolve indices on master node to make sure it doesn't hit partitions that are being deleted
-                    String index = table.ident().indexName();
-                    results.add(updateMapping(analysis.tableParameter().mappings(), index));
-                    results.add(updateSettings(parameterWithFilteredSettings, index));
+                    operations.set(3);
+                    String[] indices = getIndexNames(table, null);
```

Comment:
why not just call table.concreteIndices? you're already in if blocks that are repeated within getIndexNames